src/core/server/base.js

```diff
@@ -199,6 +199,16 @@ function baseServer(routes, createStore, { appInstanceName = appName } = {}) {
       webpackIsomorphicTools.refresh();
     }
 
+    const cacheAllResponsesFor = config.get('cacheAllResponsesFor');
+    if (cacheAllResponsesFor) {
+      log.warn(oneLine`Sending a Cache-Control header so that the client caches
+        all requests for ${cacheAllResponsesFor} seconds`);
+      res.set('Cache-Control', `public, max-age=${cacheAllResponsesFor}`);
+    }
+
+    // Vary the cache on Do Not Track headers.
+    res.vary('DNT');
```

Comment:
The browser requests the header in upper-case as `DNT: 1` but it shouldn't matter since I believe `res.vary()` accepts a case-insensitive header value. This is not documented though.